src/Haskell/Ide/Engine/MonadFunctions.hs

```diff
@@ -1,21 +1,163 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-} -- For MonadLogger IO instance
 module Haskell.Ide.Engine.MonadFunctions
   (
   -- * Logging functions
-    logm
+    withStdoutLogging
+  , withStderrLogging
+  , withFileLogging
+  , setLogLevel
+  , setLogTimeFormat
+  , logm
   , debugm
   ) where
 
-
-import           Control.Logging
 import           Control.Monad.IO.Class
+import           Control.Monad.Logger
 import qualified Data.Text as T
+import           Haskell.Ide.Engine.Monad()
 import           Prelude hiding (log)
+import Control.Exception.Lifted
+import Control.Monad
+import Control.Monad.Trans.Control
+import Data.IORef
+import Data.Monoid
+import Data.Text as T
+import Data.Time
+import Prelude hiding (log)
+import System.IO.Unsafe
+import System.Log.FastLogger
 
 -- ---------------------------------------------------------------------
 
 logm :: MonadIO m => String -> m ()
-logm s = liftIO $ log $ T.pack s
+logm s = liftIO $ logInfoN $ T.pack s
+
+debugm :: MonadIO m => String -> m ()
+debugm s = liftIO $ logDebugN $ T.pack s
+
+-- ---------------------------------------------------------------------
+
+-- instance MonadLoggerIO IO where
+instance MonadLogger IO where
+    monadLoggerLog loc src lvl msg = loggingLogger loc lvl src msg
+
+-- ---------------------------------------------------------------------
+
+{-
+From https://hackage.haskell.org/package/logging-3.0.2/docs/src/Control-Logging.html
+-}
+
+-- data LogLevel = LevelDebug | LevelInfo | LevelWarn | LevelError | LevelOther Text
+--     deriving (Eq, Prelude.Show, Prelude.Read, Ord)
+
+-- type LogSource = Text
+
+logLevel :: IORef LogLevel
+{-# NOINLINE logLevel #-}
+logLevel = unsafePerformIO $ newIORef LevelDebug
+
+-- | Set the verbosity level.  Messages at our higher than this level are
+--   displayed.  It defaults to 'LevelDebug'.
+setLogLevel :: LogLevel -> IO ()
+setLogLevel = atomicWriteIORef logLevel
+
+logSet :: IORef LoggerSet
+{-# NOINLINE logSet #-}
+logSet = unsafePerformIO $
+    newIORef (error "Must call withStdoutLogging or withStderrLogging")
+
+logTimeFormat :: IORef String
+{-# NOINLINE logTimeFormat #-}
+logTimeFormat = unsafePerformIO $ newIORef "%Y %b-%d %H:%M:%S%Q"
+
+-- | Set the format used for log timestamps.
+setLogTimeFormat :: String -> IO ()
+setLogTimeFormat = atomicWriteIORef logTimeFormat
+
+-- | This function, or 'withStderrLogging', must be wrapped around whatever
```

Comment:
While perhaps it's clear from the type, it might be good to mention in the docs that these modify global state.  So, if you `Control.Concurrent.Async.race (withStdoutLogging (logm "a")) (withFileLogging "log" (logm "b"))`, it's not deterministic which messages go to which logs.